tests/mock_singleton_helpers.h

```diff
@@ -26,27 +26,39 @@
 #include <utility>
 
 #define MP_MOCK_SINGLETON_INSTANCE(mock_class)                                                                         \
+public:                                                                                                                \
```

Comment:
For my knowledge, could you please explain why adding `public` here is necessary?  Is it that it defaults to `private` if not explicit even though this is free static function?

Reply:
With the order this is used in below, and since "includers" of this file currently use only the composite `BOILERPLATE` macro, this has no effect. However, these more elementary macros are as available as the composite one. For instance, we could use `MP_MOCK_SINGLETON_INSTANCE` in the `MockProcessFactory` today without the rest of the boilerplate.

Before this, the macros left it to the client to choose what access section to place them in. But that `academy` thing is really just an implementation detail which we should not leak into the caller's public members. Yet once we make it private, there is no way to revert to the previous access class (unless the macros received it as a parameter, which might be excessive). 

IOW, with the `private:` in line 36, the macro now potentially changed the access section at the call site. We know the new types are needed publicly, so I added that `public:` in line 40, but we can't leave the choice of access class to the caller anymore (we'd still be changing from `private:` or `protected:`). A consistent interface in all the macros requires the client to buy into the access class when using the macros, so I made that explicit by adding a public in those that remained. 

Another `public:` in the composite macro is probably in order as well, so I'll just add that in another commit, but let me know if this makes sense, or if you think I should pass `public` to the macro, or something else.

Reply:
Thanks for the detailed explanation.  That makes perfect sense and I'm good with this approach.